src/GraphQL/Execution/ExecutionNode.cs

```diff
@@ -59,6 +59,55 @@ public IObjectGraphType GetParentType(ISchema schema)
 
             return null;
         }
+
+        public IEnumerable<string> Path
+        {
+            get
+            {
+                var node = this;
+                var count = 0;
+                while (!(node is RootExecutionNode))
+                {
+                    node = node.Parent;
+                    ++count;
+                }
+
+                var pathList = new string[count];
```

Comment:
By the way, in the future you can use ArrayPool here. But there will be some difficulties in choosing where to return the array to the pool.